### ISSUE TITLE
Metrics are always enabled

### ISSUE THREAD

**Issue by dapplion** [ChainSafe/lodestar#2114]
PR https://github.com/ChainSafe/lodestar/pull/2112 adds a C++ script to collect GC metrics. This script breaks when run in a worker apparently, which happens on the multi-thread test.

However, metrics SHOULD be disabled by default and should no be loaded on the multi-thread test.

```
2021-02-28 18:15:31 [NODE 2]           warn: Low peer count peers=0
FATAL ERROR: HandleScope::HandleScope Entering the V8 API without proper locking in place
 1: 0x9dab70 node::Abort() [/opt/hostedtoolcache/node/12.14.1/x64/bin/node]
 2: 0x9dbd26 node::OnFatalError(char const*, char const*) [/opt/hostedtoolcache/node/12.14.1/x64/bin/node]
 3: 0xb3af5a v8::Utils::ReportApiFailure(char const*, char const*) [/opt/hostedtoolcache/node/12.14.1/x64/bin/node]
 4: 0xb3c4de v8::HandleScope::HandleScope(v8::Isolate*) [/opt/hostedtoolcache/node/12.14.1/x64/bin/node]
 5: 0x94a98e node::InternalCallbackScope::InternalCallbackScope(node::Environment*, v8::Local<v8::Object>, node::async_context const&, node::InternalCallbackScope::ResourceExpectation) [/opt/hostedtoolcache/node/12.14.1/x64/bin/node]
 6: 0x94b8a5 node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*, node::async_context) [/opt/hostedtoolcache/node/12.14.1/x64/bin/node]
 7: 0x7f5245046527  [/home/runner/work/lodestar/lodestar/node_modules/gc-stats/build/gcstats/v1.4.0/Release/node-v72-linux-x64/gcstats.node]
 8: 0x12d7951  [/opt/hostedtoolcache/node/12.14.1/x64/bin/node]
 9: 0x12e9f58  [/opt/hostedtoolcache/node/12.14.1/x64/bin/node]
10: 0x12d82db uv_run [/opt/hostedtoolcache/node/12.14.1/x64/bin/node]
11: 0xa1ce80 node::NodeMainInstance::Run() [/opt/hostedtoolcache/node/12.14.1/x64/bin/node]
12: 0x9acb68 node::Start(int, char**) [/opt/hostedtoolcache/node/12.14.1/x64/bin/node]
13: 0x7f524d89bbf7 __libc_start_main [/lib/x86_64-linux-gnu/libc.so.6]
14: 0x94a115  [/opt/hostedtoolcache/node/12.14.1/x64/bin/node]
Aborted (core dumped)
```

**Comment by wemeetagain** [comment]
:+1: the metrics object should be optionally passed to every top-level module.
```ts
interface IBeaconChainModules {
  db: IBeaconDb;
  ...
  metrics?: IBeaconMetrics;
}
```
All of the metrics logic can then be hidden behind a `if (metrics) {` conditional.

**Comment by dapplion** [comment]
Maybe we can find a less verbose way of not running any code at all if metrics are disabled but consumers can just do

```ts
this.metrics.someMetric.inc(1)
```

**Comment by wemeetagain** [comment]
IMO 'less verbose', ie consumers always hitting metrics, will always have a tradeoff with increased runtime cost.

If a consumer has code like `this.metrics.someMetric.set(this.expensiveFunction())`, no amount of fanciness within the metrics (eg: proxy, extended methods, etc) will stop the expensive function from running.

**Comment by wemeetagain** [comment]
Perhaps a fancy thing we could do is wrap metrics calls in anonymous functions?
```ts
this.metrics.maybe(() => this.metrics.someMetric.set(this.expensiveFunction());
```
More a matter of taste whether that or a simple `if` suffices.
```ts
if (metrics) this.metrics.someMetric.set(this.expensiveFunction());
```

**Comment by dapplion** [comment]
In the case you show above we can expose a property in metrics `enabled: boolean` to be used there only
```ts
if (this.metrics.enabled) this.metrics.someMetric.set(this.expensiveFunction();
```

**Comment by wemeetagain** [comment]
Resolved with #2117
